pkg/cmd/pipeline/describe.go

```diff
@@ -0,0 +1,134 @@
+// Copyright © 2019 The Knative Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipeline
+
+import (
+	"fmt"
+	"io"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/cli/pkg/formatted"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cliopts "k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+const (
+	resHeader  = "\nRESOURCE\nNAME\tTYPE\n"
+	resBody    = "%s\t%s\n"
+	taskHeader = "\nTASK\nNAME\tTASKREF\tRUNAFTER\n"
+	taskBody   = "%s\t%s\t%v\n"
+	runsHeader = "\nRUN\nNAME\tSTARTED\tDURATION\tSTATUS\n"
+	runsBody   = "%s\t%s\t%s\t%s\n"
+)
+
+var pname string
+
+func describeCommand(p cli.Params) *cobra.Command {
+	f := cliopts.NewPrintFlags("describe")
+
+	c := &cobra.Command{
+		Use:     "describe",
+		Aliases: []string{"desc"},
+		Short:   "Describes a pipeline in a namespace",
+		Long:    ``,
+		Args:    cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			pname = args[0]
+			if pname == "" {
```

Comment:
if `Args:    cobra.MinimumNArgs(1),` is specified then do we need this error checking?

Reply:
it only makes sense if an empty string "" is given as argument.
we can actually remove it.